Tests/xcodeprojMeasurementsTests/XcodeProjGenerator.swift

```diff
@@ -0,0 +1,144 @@
+import Foundation
+import PathKit
+@testable import xcodeproj
+
+final class XcodeProjGenerator {
+    
+    struct GenerateSetting {
+        let targetCount: Int = 5
+        let groupCountInLevel: Int = 4
+        let fileCountInLevel: Int = 10
+        let maxGroupDepth: Int = 4
+    }
+    
+    @discardableResult
+    func generate(path: Path, sourceRoot: Path, setting: GenerateSetting = GenerateSetting()) throws -> XcodeProj {
+        let workspace = XCWorkspace()
+        let buildConfigurationList = XCConfigurationList()
+        let pbxproj = PBXProj(rootObject: nil, objectVersion: 0, archiveVersion: 0, classes: [:], objects: [])
```

Comment:
I'd suggest creating the `buildConfigurationList` right after creating the `pbxproj` and then adding it to the `pbxproj` *(line 28)*